web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -228,12 +228,13 @@ export class DataStoreService {
 
   private static formToJS(form: Form): {} {
     const { fields, ...formDoc } = form;
+    let index = 0;
```

Comment:
Could you please extract the call to `reduce` into its own named method to help make it clearer that this is where indices are reassigned here based on order in the form?

General question - do we ever sort the fields by index when displaying the fields in the form editor?

Reply:
Good question, right now the forms are not being rendered in the UI and only new forms are being created. I am going to make my next PR for rendering existing forms and ability to create new form using tabular switch between forms. I will consider sorting in that PR.

Reply:
Ok, will let this through for now. Thanks for creating a bug to track.